modules/alloy/src/test/ts/browser/ui/dropdown/MenuRadioTest.ts

```diff
@@ -0,0 +1,167 @@
+  });
+
+  const hook = GuiSetup.bddSetup((store, _doc, _body) => GuiFactory.build(
+    Menu.sketch({
+      value: 'test-menu-1',
+      items: Arr.map([
+        { type: 'item', data: { value: 'alpha', meta: { text: 'Alpha' }}, hasSubmenu: false, toggling: togglingConfig(store, 'alpha') },
+        { type: 'item', data: { value: 'beta', meta: { text: 'Beta' }}, hasSubmenu: false, toggling: togglingConfig(store, 'beta') }
+      ], TestDropdownMenu.renderItem),
+      dom: {
+        tag: 'ol',
+        classes: [ 'test-menu' ]
+      },
+      components: [
+        Menu.parts.items({ })
+      ],
+
+      markers: {
+        item: TestDropdownMenu.markers().item,
+        selectedItem: TestDropdownMenu.markers().selectedItem
+      },
+
+      menuBehaviours: Behaviour.derive([
+        AddEventsBehaviour.config('menu-test-behaviour', [
+          AlloyEvents.run(MenuEvents.focus(), store.adder('menu.events.focus'))
+        ])
+      ])
+    })
+  ));
+
+  beforeEach(() => {
+    hook.store().clear();
+  });
+
+  const triggerFocusItem = (target: SugarElement<HTMLLIElement>) => {
+    AlloyTriggers.dispatch(hook.component(), target, SystemEvents.focusItem());
+  };
+
+  const itemStructure = (focused: boolean, checked: boolean): ApproxStructure.Builder<StructAssert> => (s, str, arr) =>
+    s.element('li', {
+      classes: [
+        focused ? arr.has('selected-item') : arr.not('selected-item'),
+        checked ? arr.has('checked') : arr.not('checked'),
+      ],
+      attrs: {
+        'aria-checked': str.is(String(checked)),
+        'role': str.is('menuitemradio')
+      }
+    });
+
+  it('TINY-8602: Focus first menu item and check structure', () => {
+    const component = hook.component();
+    const store = hook.store();
+    const alpha = UiFinder.findIn<HTMLLIElement>(component.element, 'li[data-value="alpha"]').getOrDie();
+
+    store.assertEq('Before focusItem event', []);
+
+    triggerFocusItem(alpha);
+    Assertions.assertStructure(
+      'After focusing item on alpha',
+      ApproxStructure.build((s, str, arr) => s.element('ol', {
+        classes: [
+          arr.has('test-menu')
+        ],
+        children: [
+          itemStructure(true, false)(s, str, arr),
+          itemStructure(false, false)(s, str, arr)
+        ]
+      })),
+      component.element
+    );
+
+    store.assertEq('After focusItem event (alpha)', [ 'menu.events.focus' ]);
+  });
+
+  it('TINY-8602: Focus second menu item and check structure', () => {
+    const component = hook.component();
+    const store = hook.store();
+    const beta = UiFinder.findIn<HTMLLIElement>(component.element, 'li[data-value="beta"]').getOrDie();
+
+    triggerFocusItem(beta);
+    Assertions.assertStructure(
+      'After focusing item on beta',
+      ApproxStructure.build((s, str, arr) => s.element('ol', {
+        classes: [
+          arr.has('test-menu')
+        ],
+        children: [
+          itemStructure(false, false)(s, str, arr),
```

Comment:
Maybe make this one argument with two fields, just do it's a bit more self-documenting. Or can you use named parameters in TS? - that would be another option.

Reply:
Yeah good point, have done that. Also no named parameters in TS unfortunately 😞